src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/builder/VisitDtoBuilder.kt

```diff
@@ -37,4 +42,17 @@ class VisitDtoBuilder {
       userType = visitEntity.userType,
     )
   }
+
+  private fun getApplicationReference(
+    visitEntity: Visit,
+  ): String {
+    val application = visitEntity.getLastCompletedApplication()
+    return application?.let {
+      it.reference
+    } ?: run {
+      // This catches an issues when two requests from the booking occur at the same time see bookVisit method in visit service
+      LOG.error("Visit ${visitEntity.reference} should have a completed application")
+      "toBeAnnounced"
```

Comment:
Will this set the application reference = "toBeAnnounced" ? Is that what we want here?